src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -397,6 +399,47 @@ public void testCategoryIndexPageTableInSyncWithAllChecksPageTable() throws Exce
         }
     }
 
+    @Test
+    public void testFiltersIndexPageTable() throws Exception {
+        final Path filtersIndexPath = Path.of("src/site/xdoc/filters/index.xml");
+        final String input = Files.readString(filtersIndexPath);
+        final Document document = XmlUtil.getRawXml(filtersIndexPath.toString(), input, input);
+        final NodeList sources = document.getElementsByTagName("tr");
+
+        for (int position = 0; position < sources.getLength(); position++) {
+            final Node tableRow = sources.item(position);
+            final Iterator<Node> cells = XmlUtil
+                    .findChildElementsByTag(tableRow, "td").iterator();
+
+            if (!cells.hasNext()) {
+                continue;
+            }
+
+            final Node nameCell = cells.next();
+            if (!cells.hasNext()) {
+                continue;
+            }
+
+            final Node descriptionCell = cells.next();
+            final String filterName = XmlUtil.sanitizeXml(nameCell.getTextContent().trim());
+            final String description = XmlUtil.sanitizeXml(descriptionCell.getTextContent().trim());
+
+            assertWithMessage("Filter name in filters/index.xml should not be empty")
+                    .that(filterName)
+                    .isNotEmpty();
+
+            assertWithMessage("Filter description for " + filterName
+                    + " in filters/index.xml should not be empty")
+                    .that(description)
+                    .isNotEmpty();
+
+            assertWithMessage("Filter description for " + filterName
+                    + " in filters/index.xml should end with a period")
+                    .that(description.charAt(description.length() - 1))
+                    .isEqualTo('.');
```

Comment:
please make filter special imlementation of  ` readSummaries() ` to do same  assert as we do for Checks.

```
assertWithMessage("The summary for check " + checkName
                        + " in the file \"" + path + "\""
                        + " should match the summary"
                        + " for this check in the file \"" + AVAILABLE_CHECKS_PATH + "\"")
                    .that(description)
                    .isEqualTo(summaries.get(checkName));
```